parity/cli/mod.rs

```diff
@@ -564,6 +564,10 @@ usage! {
 			"--ipc-path=[PATH]",
 			"Specify custom path for JSON-RPC over IPC service.",
 
+			ARG arg_ipc_chmod: (u16) = 666u16, or |c: &Config| c.ipc.as_ref()?.chmod.clone(),
```

Comment:
It seems odd to me to store `the  wrong` value in an integer for this, if the number gets used elsewhere.

The `CLI parser` doesn't understand `0o666` right?

Then, I suggest:
1. a wrapper type over `u16` to indicate this 
2. store the `cli arg` as a string and parse it in `start_ipc`

Reply:
well, the syntax "0o666" is pretty rust specific.

when you use `chmod` program on linux you don't specify the permissions as 0o666.
Based on this, and the explicit feature request: https://github.com/paritytech/parity-ethereum/issues/9703#issuecomment-446054650

I feel like it's better if we don't complicate the user experience.

Reply:
I kind of agree with @niklasad1 here: it seems like a bad idea to take any value and pipe it through into the networking layer with no validation, trusting the OS to validate the user input. If `0o666` is too unpalatable for users then perhaps keeping it a string and do the validation before calling `set_mode()` seems better.

Reply:
I think `chmod` actually accepts the permission arguments as strings (it's just in bash you don't put it in quotes, but it gets converted to a string). It may make sense to do the same, but OTOH having `u16` does the string validation for us in the cli. Maybe it's time to migrate our cli to stuctopt? (but that's out of scope of the PR)

Reply:
> I kind of agree with @niklasad1 here: it seems like a bad idea to take any value and pipe it through into the networking layer with no validation, trusting the OS to validate the user input. If 0o666 is too unpalatable for users then perhaps keeping it a string and do the validation before calling set_mode() seems better.


what kind of validation are we talking about here?

Reply:
validation: what stops a user from passing `1234` here? I am probably missing something but my point is: we shouldn't pass invalid values all the way down to `set_mode()` and trust that it fails.

Reply:
`1234` is actually valid linux/unix permission code.

http://www.filepermissions.com/file-permissions-index

Reply:
Yes Seun, FWIW specifying permissions is valid for 1-4 octal numbers otherwise it will be an error AFAIK.

However, my point is still that the `wrong value` is used in `u16`, thus it should be not possible to access it as an `u16`,  by either a Wrapper type or a String IMHO

Example wrapper:

```
struct IpcPermission {
   /// value stored as octal which can't viewed except as `octal string`
   chmod: u16
}

impl ToString for IpcPermission {...}

impl IpcPermission {
   fn new(chmod: u16) -> Self {...}
}

Reply:
ok so if I understand correctly, you want to limit the number of digits a user can supply as the chmod value to 4. 

My argument to that would be that I feel like if you're running an ethereum node you kind of already should be familiar with octal permissions but for the sake of defensive programming it's also a great feature to have.

Reply:
> ok so if I understand correctly, you want to limit the number of digits a user can supply as the chmod value to 4.

No, I just tried to explain that valid chmod values are x, xx, xxx, xxxx were x is value between 0 and 7 and I think we should respect the user input. If the current user privilege level  running parity is lower than the supplied then we should rely on the OS to take care of that IMO.